### PR TITLE
chore: automatic changelog releases

### PR DESCRIPTION
**Pull request by icazevedo**
## What's the purpose of this pull request?

This PR introduces automatic versioning for changelog and `package.json`.

## How does it work?

Every time a PR is merged, a new commit is created bumping up the version of `package.json` and adding the required changelog entries under `CHANGELOG.md`.

## How to test it?

Merge a PR in the main branch 😝 

## References

https://github.com/conventional-changelog/standard-version

### CODE REVIEW

**Review by filipewl — COMMENTED** [review]
This will save us a lot of time! 

Previously, because we needed to manually include an entry to the change log with the PR #, that meant we needed to push the commit, open the PR, pick the #, and push another commit to update the change log. That extra commit meant more waiting for the CI...


Left a couple of comments for us to discuss first, but overall this looks good to me!

**Review by emersonlaurentino — APPROVED** [review]
GJ

File: .github/workflows/release.yml
Comment (icazevedo) [reply .github/workflows/release.yml:10]: hmmmm good point! I'll see what I can do

File: package.json
Comment (filipewl) [reply package.json:23]: I think it was to express the current closed beta status.

Comment (icazevedo) [reply package.json:23]: I get it. Let me see if that's it

Comment (icazevedo) [reply package.json:23]: I couldn't achieve what we had before. Release-it creates another "number" for the beta modifier, like `22.30.0-beta.0` and keeps incrementing it (`22.30.0-beta.1`) instead of using the default rule (calendar) and bump it to something like `22.30.1-beta.0`

Comment (filipewl) [reply package.json:23]: Aaah, okay. Thanks for trying!

File: CHANGELOG.md
Comment (filipewl) [inline CHANGELOG.md:8]: Are these removals of the unreleased intentional? I remember seeing this file in the new format, but I think that's gone now after some new commits.
🤔

File: .github/workflows/release.yml
Comment (icazevedo) [reply .github/workflows/release.yml:10]: I was able to do it! :D

File: CHANGELOG.md
Comment (icazevedo) [reply CHANGELOG.md:8]: It is intentional, yes. The changelog is generated automatically, so there's no need for an unreleased section. In the first commits we were using a different library (standard-version) that required this section. Release-it works great without it.

Comment (icazevedo) [reply CHANGELOG.md:8]: wait, now I get it. I'll release a new version in a sec, this will go away hahaha

Comment (icazevedo) [reply CHANGELOG.md:8]: They'll be added once we merge this PR

**Review by filipewl — APPROVED** [review]
In Ícaro we trust!

**Pull request merged**
